### ISSUE TITLE
Custom node breaks Save File Formatting

### ISSUE THREAD

**Issue by dexeonify** [pythongosssss/ComfyUI-Custom-Scripts#191]
## Steps to Reproduce
1. Launch a clean instance of ComfyUI without any custom nodes.
2. Create a simple workflow that loads an image and save the file formatted as `%date:yyyy-MM-dd%`, using two methods:
a) Using `filename_prefix` directly
![image](https://github.com/pythongosssss/ComfyUI-Custom-Scripts/assets/72267349/879a16cb-b4c3-49ff-9ac7-da925acbe359)
b) Connect a primitive node after converting `filename_prefix` to input
![image](https://github.com/pythongosssss/ComfyUI-Custom-Scripts/assets/72267349/c08b5061-5a86-4197-86b3-3795ca2c4683)
4. Check the output folder.
5. Install `ComfyUI-Custom-Scripts` and relaunch ComfyUI, then retry the two workflows above.

## Expected result
Both methods should work just like vanilla ComfyUI, or at least print a warning/error somewhere.

## Actual result
With the custom node installed, the first method works flawlessly, while the second method silently broke. If you checked the output folder, an incomplete file `%date` is created. No console messages was logged by Custom-Scripts or ComfyUI.

### Notes
- I've read the README and I don't see anything that mentions this change in behaviour of the save image and primitive node.
- I'm surprised this wasn't reported considering it breaks a core functionality, are there some other issues at play here?

**Comment by pythongosssss** [comment]
Have you set "Run widget replace on values" to true on the primitive node?
The default on Primitives is off, so replacements are not run, whereas it always runs on the SaveImage node.

![image](https://github.com/pythongosssss/ComfyUI-Custom-Scripts/assets/125205205/0d6e8d51-d8a1-4797-85f8-e78eb9620c8d)

If I enable that, it works fine (but there is a bug with the primitive node reloading these values when you reload the workflow)

**Comment by dexeonify** [comment]
Yes, it works fine with that option enabled. Now I wonder how/why does vanilla ComfyUI perform replacements without that option enabled. To further complicate the matter, some users have reported that it doesn't work even without this custom node installed (comfyanonymous/ComfyUI#2832).

**Comment by Delerium** [comment]
I'd like to add that by setting "Run widget replace on values" to true, any time you pull this image into ComfyUI, it will not retain your custom save file formatting, as it will be overwritten with the hard coded result your formatting generated the first time.  This is not desired at all.  Any way around this?

**Comment by dexeonify** [comment]
> If I enable that, it works fine (but there is a bug with the primitive node reloading these values when you reload the workflow)

It's a known bug.
The workaround, or should I say the proper method, is to use the first method and forego using primitive node entirely.

**Comment by NPDX909** [comment]
This is still a bug.
